### PR TITLE
limiting GD view based on course enrollment

### LINKED ISSUE #1288: Add configuration option for administrators to restrict views available based on number of students in class
### Describe your problem or feature you'd like added
For small classes, maintaining student privacy becomes a concern with data visualizations. This becomes most important for the Grade Distribution visualization. In a small class, there is a risk that students could deduce an individual student's grade based on the histogram data.

### Describe the solution you'd like
Provide a way for MyLA administrators to limit which views are available to be added to courses based on a minimum number of students in the course.  For example, allow administrators to set a threshold of 11 students, indicating that courses with 11 or fewer students would not be able to enable Grade Distribution.  

### Describe any possible alternatives you've considered
Currently, institutions are making recommendations to instructors via documentation to let them know what minimum enrollment should be considered when turning on views.  Adding an administrative configuration would allow institutions to enforce enrollment thresholds for the Grade Distribution view.

Further discussion may be needed to determine whether the threshold needs to apply to only the Grade Distribution view or if it should also apply to the Resources Accessed view.

### PR DESCRIPTION
**Pull request by pushyamig**
Fixes #1288

### CODE REVIEW

**Comment by pushyamig** [comment]
@zqian opened a new PR with same changes. I don't know why you cannot checkout the previous PR locally before.

**Comment by zqian** [comment]
This works for me. Reviewing now! Thanks!

File: dashboard/views.py [hunk 1]
```diff
@@ -460,7 +460,10 @@ def grade_distribution(request, course_id=0):
             'course_id': course_id,
             'enrollment_type': 'StudentEnrollment'
         })
-    if df.empty or df.count().current_grade < 6:
+    if len(df) <= config.GRADE_DISTRIBUTION_MINIMUM:
+        logger.error(f"Course enrollment count {len(df)} is lower than the minimum required to display the Grade Distribution visualization {config.GRADE_DISTRIBUTION_MINIMUM}")
```
Comment (zqian) [inline dashboard/views.py:464]: Can define a message string to be used for both line 464 and 465.

File: dashboard/views.py [hunk 2]
```diff
@@ -460,7 +460,10 @@ def grade_distribution(request, course_id=0):
...
+        return HttpResponse(json.dumps({'gd_disable':'true','gd_msg': f'Grade Distribution view is disabled because the course enrollment is less than {config.GRADE_DISTRIBUTION_MINIMUM}'}), content_type='application/json')
+    if df.empty or df.count().current_grade < 6 :
```
Comment (zqian) [inline dashboard/views.py:466]: is the check for 6 still necessary, given the setting of `config.GRADE_DISTRIBUTION_MINIMUM`?

If so, could we make the number 6 as another admin config variable?

Comment (pushyamig) [reply dashboard/views.py:466]: 10 is the number we set as default. But some universities set it 0 (with out knowingly) then the Grade distribution need atleast 6 grades for rest of the depersonalized logic to work. So it need to check for GD to work. 
This is not something new I introduced with this change [it was there from before](https://github.com/tl-its-umich-edu/my-learning-analytics/blob/master/dashboard/views.py#L463). 
I don't feel this needs another configuration value.

File: dashboard/views.py (hunk 1 above)
Comment (pushyamig) [reply dashboard/views.py]: OK, I have changed as requested 78eb5635a161fbca8e9143fb2c391879afb7d485

File: dashboard/views.py (hunk 2 above)
Comment (zqian) [reply dashboard/views.py:467]: I see. 

Can we expose the number 6 to the configuration description text? Something like `Minimum number of student enrollments required to show the Grade Distribution visualization. The number needs to be greater than 6 for depersonalized logic to work in the Grade Distribution view. `. WDYT?

Comment (pushyamig) [reply dashboard/views.py:467]: Ok, I can do that.

Comment (pushyamig) [reply dashboard/views.py]: Done

**Review by zqian — APPROVED** [review]

**Pull request merged**